TS infer type for class static field with error

<!-- 🚨 STOP 🚨 STOP 🚨 STOP 🚨

Half of all issues filed here are duplicates, answered in the FAQ, or not appropriate for the bug tracker. Even if you think you've found a *bug*, please read the FAQ first, especially the Common "Bugs" That Aren't Bugs section!

Please help us by doing the following steps before logging an issue:
  * Search: https://github.com/Microsoft/TypeScript/search?type=Issues
  * Read the FAQ: https://github.com/Microsoft/TypeScript/wiki/FAQ

Please fill in the *entire* template below.
-->

<!--
Please try to reproduce the issue with the latest published version. It may have already been fixed.
For npm: `typescript@next`
This is also the 'Nightly' version in the playground: http://www.typescriptlang.org/play/?ts=Nightly
-->
**TypeScript Version:**  3.9.7

<!-- Search terms you tried before logging this (so others can find this issue more easily) -->
**Search Terms:**

**Code**

```ts
// file a.ts
export class A {
    static foo() {}
}

// file b.ts
import { A } from './a';
export class B extends A {}

// file index.ts
import { B } from './b';
export const foo = B.foo;
```

**Expected behavior:**
output declaretion files with no errors.

**Actual behavior:**
```ts
// in index.d.ts
export declare const foo: typeof import("./a").A.foo;
// Namespace '"<build_path>/a"' has no exported member 'A'.ts(2694)
```

**Playground Link:** <!-- A link to a TypeScript Playground "Share" link which demonstrates this behavior -->
https://github.com/Frezc/classstatictypeinfer-repro

**Related Issues:** <!-- Did you find other bugs that looked similar? -->

@weswigham we build these .d.ts files
a.d.ts
```ts
export declare class A {
    static foo(): void;
}
```
b.d.ts
```ts
import { A } from './a';
export declare class B extends A {
}
```
index.d.ts
```ts
export declare const foo: typeof import("./a").A.foo;
```
which raises an error if you try to build the .d.ts files together:
```
index.d.ts:1:48 - error TS2694: Namespace '"C:/Throwaway/classstatictypeinfer-repro/build/a"' has no exported member 'A'.

1 export declare const foo: typeof import("./a").A.foo;
                                                 ~

```

--- a/src/compiler/checker.ts
+++ b/src/compiler/checker.ts
@@ -14349,7 +14349,10 @@ namespace ts {
                         // typeof a.b.c is normally resolved using `checkExpression` which in turn defers to `checkQualifiedName`
                         // That, in turn, ultimately uses `getPropertyOfType` on the type of the symbol, which differs slightly from
                         // the `exports` lookup process that only looks up namespace members which is used for most type references
-                        const next = node.isTypeOf ? getPropertyOfType(getTypeOfSymbol(resolveSymbol(currentNamespace)), current.escapedText) : getSymbol(getExportsOfSymbol(getMergedSymbol(resolveSymbol(currentNamespace))), current.escapedText, meaning);
+                        const mergedResolvedSymbol = getMergedSymbol(resolveSymbol(currentNamespace));
+                        const next = node.isTypeOf
+                            ? getPropertyOfType(getTypeOfSymbol(mergedResolvedSymbol), current.escapedText)
+                            : getSymbol(getExportsOfSymbol(mergedResolvedSymbol), current.escapedText, meaning);
                         if (!next) {
                             error(current, Diagnostics.Namespace_0_has_no_exported_member_1, getFullyQualifiedName(currentNamespace), declarationNameToString(current));
                             return links.resolvedType = errorType;

--- a/src/compiler/checker.ts
+++ b/src/compiler/checker.ts
@@ -14346,7 +14346,10 @@ namespace ts {
                     let current: Identifier | undefined;
                     while (current = nameStack.shift()) {
                         const meaning = nameStack.length ? SymbolFlags.Namespace : targetMeaning;
-                        const next = getSymbol(getExportsOfSymbol(getMergedSymbol(resolveSymbol(currentNamespace))), current.escapedText, meaning);
+                        // typeof a.b.c is normally resolved using `checkExpression` which in turn defers to `checkQualifiedName`
+                        // That, in turn, ultimately uses `getPropertyOfType` on the type of the symbol, which differs slightly from
+                        // the `exports` lookup process that only looks up namespace members which is used for most type references
+                        const next = node.isTypeOf ? getPropertyOfType(getTypeOfSymbol(resolveSymbol(currentNamespace)), current.escapedText) : getSymbol(getExportsOfSymbol(getMergedSymbol(resolveSymbol(currentNamespace))), current.escapedText, meaning);
                         if (!next) {
                             error(current, Diagnostics.Namespace_0_has_no_exported_member_1, getFullyQualifiedName(currentNamespace), declarationNameToString(current));
                             return links.resolvedType = errorType;

--- a/tests/baselines/reference/importTypeTypeofClassStaticLookup.symbols
+++ b/tests/baselines/reference/importTypeTypeofClassStaticLookup.symbols
@@ -1,0 +1,14 @@
+=== tests/cases/compiler/a.d.ts ===
+export declare class A {
+>A : Symbol(A, Decl(a.d.ts, 0, 0))
+
+    static foo(): void;
+>foo : Symbol(A.foo, Decl(a.d.ts, 0, 24))
+}
+
+=== tests/cases/compiler/index.d.ts ===
+export const foo: typeof import("./a").A.foo;
+>foo : Symbol(foo, Decl(index.d.ts, 0, 12))
+>A : Symbol(A, Decl(a.d.ts, 0, 0))
+>foo : Symbol(A.foo, Decl(a.d.ts, 0, 24))
+

--- a/tests/baselines/reference/importTypeTypeofClassStaticLookup.types
+++ b/tests/baselines/reference/importTypeTypeofClassStaticLookup.types
@@ -1,0 +1,14 @@
+=== tests/cases/compiler/a.d.ts ===
+export declare class A {
+>A : A
+
+    static foo(): void;
+>foo : () => void
+}
+
+=== tests/cases/compiler/index.d.ts ===
+export const foo: typeof import("./a").A.foo;
+>foo : () => void
+>A : any
+>foo : any
+

--- a/tests/cases/compiler/importTypeTypeofClassStaticLookup.ts
+++ b/tests/cases/compiler/importTypeTypeofClassStaticLookup.ts
@@ -1,0 +1,7 @@
+// @filename: a.d.ts
+export declare class A {
+    static foo(): void;
+}
+
+// @filename: index.d.ts
+export const foo: typeof import("./a").A.foo;

